java/ql/src/semmle/code/java/frameworks/jackson/JacksonSerializability.qll

```diff
@@ -135,6 +156,21 @@ class JacksonDeserializableField extends DeserializableField {
   }
 }
 
+/** A call to a field that may be deserialized using the Jackson JSON framework. */
+class JacksonDeserializableFieldAccess extends FieldAccess {
+  JacksonDeserializableFieldAccess() { getField() instanceof JacksonDeserializableField }
+}
+
+/**
+ * When an object is deserialized by the Jackson JSON framework using a tainted input source,
+ * the fields that the framework deserialized are themselves tainted input data.
+ */
+class JacksonDeserializedTaintStep extends AdditionalTaintStep {
+  override predicate step(DataFlow::Node node1, DataFlow::Node node2) {
+    node2.asExpr().(JacksonDeserializableFieldAccess).getQualifier() = node1.asExpr()
```

Comment:
This should use `DataFlow::getFieldQualifier(FieldAccess)` rather than `FieldAccess.getQualifier()`. The latter is purely syntactic and only gets results if there's an actual AST element, whereas the former also gets you data-flow nodes such as implicit `this`-accesses.

Reply:
Why does the `getQualifier` method just work for `MethodAccess` then?

Reply:
It also doesn't fully work. That too only works when the qualifier is an explicit AST element. For calls, the correct thing to use is the similar `DataFlow::getInstanceArgument(Call)`.